Create / clear user cookie

joeylouise opened the pull request:
This PR:

Requires cookie-parser module
Adds dummy logout button with styling
Create cookie when user is authenticated
Clear cookie when user logs out
Relates #94

joeylouise committed:
Add logout route which deletes the cookie and redirects to the sign-up page
Relates #94

joeylouise committed:
Require cookie-parser in app.js in order to use cookies
Relates #94

joeylouise committed:
Add cookie-parser as a dependency in package.json
Relates #94

joeylouise committed:
Add dummy logout button to make sure functionality for clearing cookies works
Relates #94

joeylouise committed:
Add css for dummy logout button
Relates #94

joeylouise committed:
Set cookie in login route if user is authenticated
Relates #94

joeylouise committed:
Add cookie-parser to router and add /logout route
Relates #94

des-des commented on src/server/controllers/login.js:
```diff
@@ -19,8 +19,7 @@ module.exports = (req, res) => {
       if (isAuthenticated) {
-        // need to set cookie
-        res.redirect('/');
+        res.cookie('name', login, { maxAge: 3600000 }).redirect('/');
```
Is this one hour?

I would write it as 1000 * 60 * 60, otherwise it is hard to know what it means

des-des reviewed (changes requested):
Okay, right now this is not secure, you have no way of verifying that a cookie returned sent by a client is authentic. I would use the cookie-session npm module. This will sign and verify your cookie.

joeylouise committed:
Merge branch 'master' into create-user-cookie

joeylouise committed:
Require cookie-session in app.js
Relates #94

joeylouise committed:
Require cookie-session in router and create session for authenticated
Relates #94

joeylouise committed:
Update first parameter that getMemories takes so that it takes in user"s login
relates #94

joeylouise committed:
Set req.session.name as user login instead of setting res.cookie
Relates #94

joeylouise committed:
Clear session when user clicks logout
Relates #94

joeylouise committed:
Require cookie-session as a dependency in package.json
Relates #94;

joeylouise committed:
Merge branch 'master' into create-user-cookie

joeylouise commented:
@des-des I've changed this to use cookie-session and have set req.session.name when user logs in.

I've added in the session in `router.js` because it doesn't work when I add it in `app.js`

joeylouise committed:
Merge branch 'master' into create-user-cookie

des-des commented on src/server/router.js:
```diff
@@ -8,7 +9,23 @@ router.use(bodyParser.urlencoded({
   extended: true,
 }));
 
+const sessionSecret = '3294r4whasuodifhd13urde80sufj138ryt9-gu538re9ghvisujela4hrsf4rue5r89';
```
No! Do not commit this. Put it in the env file!

des-des commented on package.json:
```diff
@@ -32,6 +32,8 @@
   },
   "dependencies": {
     "body-parser": "^1.17.2",
+    "cookie-parser": "^1.4.3",
```
Do you need this?

des-des reviewed (approved):
coolio looks good

des-des reviewed (changes requested):
Ops see review above

joeylouise committed:
Remove cookie-parser dependency as we are now using cookie-session
Relates #94

joeylouise committed:
Remove sessionSecret variable from router.js and put in config.env
Relates #94

joeylouise committed:
Merge branch 'create-user-cookie' of github.com:pbywater/echo into create-user-cookie

joeylouise committed:
Change logout route to redirect to /login on logout
Relates #94

joeylouise committed:
Merge branch 'master' into create-user-cookie
